src/main/java/uk/gov/ida/verifyserviceprovider/configuration/HubMetadataConfiguration.java

```diff
@@ -59,8 +57,17 @@ public String getExpectedEntityId() {
 
     @Override
     public KeyStore getTrustStore() {
-        return validateTruststore(ofNullable(trustStoreConfiguration)
-                .orElseGet(() -> new DefaultHubTrustStoreConfiguration(environment)).getTrustStore());
+        return ofNullable(trustStoreConfiguration).map(TrustStoreConfiguration::getTrustStore).orElseGet(environment::getMetadataTrustStore);
+    }
+
+    @Override
+    public Optional<KeyStore> getHubTrustStore() {
+        return of(ofNullable(hubTrustStoreConfiguration).map(TrustStoreConfiguration::getTrustStore).orElseGet(environment::getHubTrustStore));
+    }
+
+    @Override
+    public Optional<KeyStore> getIdpTrustStore() {
+        return of(ofNullable(idpTrustStoreConfiguration).map(TrustStoreConfiguration::getTrustStore).orElseGet(environment::getHubTrustStore));
```

Comment:
Can we replace `environment::getHubTrustStore` with `environment::getIdpTrustStore` for `getIdpTrustStore()` method.